src/utils/ProtectedRoute.jsx

```diff
@@ -19,8 +23,19 @@ import { auth, getCurrentUser } from './auth_utils';
  * @returns The relevant path to redirect the user to dependending on authentication state.
  */
 const ProtectedRoute = ({ Component, redirectPath, roles, cookies }) => {
-  const currentUser = getCurrentUser(auth);
-  if (currentUser && roles.includes(cookies.get(cookieKeys.ROLE))) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+
+  useEffect(async () => {
+    const authenticated = await userIsAuthenticated(roles, cookies);
+    setIsAuthenticated(authenticated);
+    setIsLoading(false);
```

Comment:
since you need a try and catch cause of `getCurrentUser` in the `RefreshToken` function, here is a good place to add it. so if there is an error, then have the `isAuthenticated` to be false. then depending on the NPO project, they may add the error to some banner. also in the catch, remove all cookies that may have been added before (`clearCookies` from the cookie utils files)